pyvista/plotting/qt_plotting.py

```diff
@@ -756,6 +756,9 @@ def close(self):
         close the plotter through `signal_close`.
 
         """
+        if hasattr(self, "render_timer"):
+            self.render_timer.stop()
+        BasePlotter.close(self)
```

Comment:
I'd call `super().close()` here. `BackgroundPlotter` inherits `QtInteractor`, whose close method already calls `BasePlotter.close(self)` and stops the timer (without the check though).

Reply:
The thing is `QtInteractor.close()` calls `QVTKRenderWindowInteractorAdapter.close()` which is already done by `app_window.close()`

Reply:
I would prefer avoid unnecessary/duplicated calls to `close` since I can't guarantee the behaviour of the app after mulitple calls to `QVTKRenderWindowInteractor.close()`

Reply:
We can always make it nice later, let's first make it work

Reply:
I am probably just approaching this differently, as I don't know this classes at all and are thus mostly trying to understand how they connect.

The way I understand it QtInteractor should be closed through the signal at https://github.com/pyvista/pyvista/blob/1ed70a2bd1e0056072c78021f1c5a41dabcecd8f/pyvista/plotting/qt_plotting.py#L673. I now checked and that signal is never emitted when the window is closed programmatically in test_background_plotting_close (or at least the connected lambda isn't executed before the test stops). If it were, the changes in this PR wouldn't be needed.

I made a suggestion how to close using base classes (also passes tests). I added a safeguard that close is only called once (however if the behaviour I described above is intended, i.e. event only emitted when the window is closed externally, it's probably not needed - better safe than sorry though). Let me know what you think please:  
https://github.com/pyvista/pyvista/compare/close_event...imsodin:close_event

Reply:
This entire function (`_close`) is not needed anymore when actually emitting `signal_closed`, as that's all done in the connected `QtInteractor.close` method.

Reply:
I know it's in the ideas list and you suggested to put it into a followup PR, and that's fine with me, I just wanted to reiterate as a lot changed since the last comments happened and I might not have expressed myself understandably.  
In my opinion this is not a question of niceness, but robustness: This method duplicates code that exists in the parent `QtInteractor.close` method, thus I think that should be called (as it was before through the `app_window` hook).